base_classes/nyaml/NXbeam_optical.yaml

```diff
@@ -0,0 +1,86 @@
+doc: |
+  Properties of an optical beam at a given location.
+category: base
+(NXbeam)NXbeam_optical:
+  incident_polarization(NX_NUMBER):
```

Comment:
Personally I would tend to move this to NXbeam_polarization and then add in this polarization base class the method whereby that polarization is described/parameterized e.g.
descriptor(NXCHAR):
   enumeration: [stokes]